File: contains_duplicate/main_test.go
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type TestCase struct {
	testName       string
	input          []int
	expectedOutput bool
}

var testCases = []TestCase{
	{
		testName:       "Test1",
		input:          []int{1, 2, 3, 1},
		expectedOutput: true,
	},
	{
		testName:       "Test2",
		input:          []int{1, 2, 3, 4},
		expectedOutput: false,
	},
	{
		testName:       "Test3",
		input:          []int{1, 1, 1, 3, 3, 4, 3, 2, 4, 2},
		expectedOutput: true,
	},
}

func TestSolutionWithHashMap(t *testing.T) {
	assert := assert.New(t)

	for _, testCase := range testCases {
		t.Run(testCase.testName, func(t *testing.T) {
			input := testCase.input
			expectedOutput := testCase.expectedOutput
			output := solutionWithHashMap(input)
			assert.Equal(expectedOutput, output)
		})
	}
}

func TestSolutionWithArray(t *testing.T) {
	assert := assert.New(t)

	for _, testCase := range testCases {
		t.Run(testCase.testName, func(t *testing.T) {
			input := testCase.input
			expectedOutput := testCase.expectedOutput
			output := solutionWithArray(input)
			assert.Equal(expectedOutput, output)
		})
	}
}

File: contains_duplicate/main_test.py
# Execute: python3 -m unittest -v main_test.py

from main import Solution

import unittest

class TestSolution(unittest.TestCase):

    def test_1(self):
        s = Solution()
        input = [1,2,3,1]
        expected_output = True
        self.assertEqual(s.containsDuplicate(input), expected_output)

    def test_2(self):
        s = Solution()
        input = [1,2,3,4]
        expected_output = False
        self.assertEqual(s.containsDuplicate(input), expected_output)

    def test_3(self):
        s = Solution()
        expected_output = True
        input = [1,1,1,3,3,4,3,2,4,2]
        self.assertEqual(s.containsDuplicate(input), expected_output)

if __name__ == '__main__':
    unittest.main()